xds/src/main/java/io/grpc/xds/XdsClusterResource.java

```diff
@@ -169,9 +174,50 @@ static CdsUpdate processCluster(Cluster cluster,
     updateBuilder.filterMetadata(
         ImmutableMap.copyOf(cluster.getMetadata().getFilterMetadataMap()));
 
+    try {
+      ImmutableMap<String, Object> parsedFilterMetadata =
+          parseClusterMetadata(cluster.getMetadata());
+      updateBuilder.parsedMetadata(parsedFilterMetadata);
+    } catch (InvalidProtocolBufferException e) {
+      throw new ResourceInvalidException("xDS filter metadata invalid.");
```

Comment:
This is a woefully unhelpful error message and it is completely throwing away the exception `e`.